packages/testkit-backend/src/request-handlers.js

```diff
@@ -241,6 +241,14 @@ export function ResultConsume (_, context, data, wire) {
   const { resultId } = data
   const result = context.getResult(resultId)
 
+  if (('recordIt' in result)) {
+    return result.recordIt.return().then(summary => {
+      if (summary.value) {
+        summary = summary.value
+      }
+      wire.writeResponse(responses.Summary({ summary }, { binder: context.binder }))
+    }).catch(e => wire.writeError(e))
+  }
   return result.summary().then(summary => {
     wire.writeResponse(responses.Summary({ summary }, { binder: context.binder }))
   }).catch(e => wire.writeError(e))
```

Comment:
```suggestion
  let summaryPromise = 'recordIt' in result
    ? (async () => {return (await result.recordIt.return()).value})()
    : result.summary()
  return summaryPromise.then(summary => {
    wire.writeResponse(responses.Summary({ summary }, { binder: context.binder }))
  }).catch(e => wire.writeError(e))
```
Alternative to my above suggestion to DRY up the code (not repeating the serialization part).